src/main/java/com/streamr/broker/kafka/KafkaRecordTransformer.java

```diff
@@ -1,20 +1,28 @@
 package com.streamr.broker.kafka;
 
 import com.streamr.broker.StreamrBinaryMessage;
-import com.streamr.broker.StreamrBinaryMessageWithKafkaMetadata;
+import com.streamr.broker.StreamrBinaryMessageFactory;
+import com.streamr.client.protocol.message_layer.StreamMessage;
+import com.streamr.client.protocol.message_layer.StreamMessageFactory;
 import org.apache.kafka.clients.consumer.ConsumerRecord;
 
+import java.io.IOException;
 import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.Map;
 
 class KafkaRecordTransformer {
 	private final Map<String, Long> offsetByStream = new HashMap<>();
 
-	StreamrBinaryMessageWithKafkaMetadata transform(ConsumerRecord<String, byte[]> record) {
-		StreamrBinaryMessage streamrBinaryMessage = StreamrBinaryMessage.from(ByteBuffer.wrap(record.value()));
+	StreamMessage transform(ConsumerRecord<String, byte[]> record) throws IOException {
+		String json = new String(record.value(), StandardCharsets.UTF_8);
```

Comment:
Danger of getting an exception here (?), if the content is binary which doesn't happen to be representable in UTF-8. (Sure, all UTF-8 strings have a binary representation, but not all binary might be valid UTF-8 strings!)

Would be safer to inspect the raw byte(s) of the `byte[]` to establish whether it's a `StreamrBinaryMessage` or a JSON message, and only parse to UTF-8 String if the latter.